oarepo_oidc_einfra/resources.py

```diff
@@ -6,50 +6,186 @@
 
+if TYPE_CHECKING:
+    from werkzeug import Response
+
+log = logging.getLogger(__name__)
+
+
 upload_dump_action = action_factory("upload-oidc-einfra-dump")
 
 
-class OIDCEInfraResourceConfig(ResourceConfig):
+class OIDCEInfraUIResourceConfig(ResourceConfig):
     """Configuration for the REST API."""
 
-    blueprint_name = "oarepo_oidc_einfra"
+    blueprint_name = "oarepo_oidc_einfra_ui"
     """Blueprint name."""
 
-    url_prefix = "/oidc-einfra"
-    """URL prefix for the resource, will be at /api/oidc-einfra."""
+    url_prefix = "/auth/oidc/einfra"
+    """URL prefix for the resource."""
 
-    routes = {"upload-dump": "/dumps/upload"}
+    routes = {
+        "accept-invitation": "/invitations/<request_id>/accept",
+    }
     """Routes for the resource."""
 
 
-class OIDCEInfraResource(Resource):
+class OIDCEInfraUIResource(Resource):
     """REST API for the EInfra OIDC."""
 
-    def __init__(self, config=None):
+    def __init__(self, config: Optional[OIDCEInfraUIResourceConfig] = None):
         """Initialize the resource."""
-        super(OIDCEInfraResource, self).__init__(
-            config=config or OIDCEInfraResourceConfig()
-        )
+        super().__init__(config=config or OIDCEInfraUIResourceConfig())
 
-    def create_url_rules(self):
+    def create_url_rules(self) -> list[dict]:
+        """Create URL rules for the resource."""
+        routes = self.config.routes
+        return [
+            route("GET", routes["accept-invitation"], self.accept_invitation),
+        ]
+
+    @fresh_login_required
+    def accept_invitation(self) -> Response:
+        """Accept an invitation to join a community.
+
+        This is an endpoint to which user is directed
+        after clicking the link in the invitation email, accepting the terms and conditions and
+        accepting the invitation.
+
+        We expect the url to contain the request_id of the invitation request that was sent to the user
+        and use it to accept the invitation.
+
+        Note:
+        If user accepts the invitation but this endpoint is not called, the invitation will be forever
+        in the submitted state (until expiration). The user will still be able to access the community
+        because the AAI will return the correct capabilities for the user.
+
+        Currently, the PERUN api does not return the ID of the created invitation, so we cannot store it
+        and check in a background task if the invitation was accepted and then change the state of the request.
+
+        """
+        assert request.view_args is not None
+
+        request_id = decrypt(request.view_args["request_id"])
+
+        # force re-authentication
+        if "fresh_login_token" not in request.args:
+            # force logout and redirect to the same page with the fresh_login_token
+            # so that the user is logged in again
+            logout_user()
+            fresh_login_token = encrypt("fresh_login_token_" + str(request_id))
```

Comment:
would add (de)-base64'ing for url safety of encrypted token

Reply:
Fernet.encrypt performs `base64.urlsafe_b64encode`